src/main/java/teammates/storage/search/SearchManager.java

```diff
@@ -17,83 +29,263 @@
-        throw new SearchNotImplementedException();
+
+        QueryResponse response = null;
+        SolrQuery instructorQuery = new SolrQuery();
+
+        String cleanQueryString = cleanSpecialChars(queryString);
+        instructorQuery.setQuery(cleanQueryString);
+
+        instructorQuery.setStart(START_INDEX);
+        instructorQuery.setRows(NUM_OF_RESULTS);
+
+        try {
+            response = client.query(INSTRUCTOR_COLLECTION_NAME, instructorQuery);
+        } catch (SolrServerException e) {
+            log.severe(String.format(ERROR_SEARCH_DOCUMENT, queryString, e.getRootCause())
+                    + TeammatesException.toStringWithStackTrace(e));
+        } catch (IOException e) {
+            log.severe(String.format(ERROR_SEARCH_DOCUMENT, queryString, e.getCause())
+                    + TeammatesException.toStringWithStackTrace(e));
+        }
+
+        return InstructorSearchDocument.fromResponse(response);
     }
 
     /**
      * Batch creates or updates search documents for the given instructors.
      */
     public void putInstructorSearchDocuments(InstructorAttributes... instructors) {
-        if (!isSearchServiceActive()) {
-            log.warning("Search service is not implemented");
+        if (client == null) {
+            log.warning(ERROR_SEARCH_NOT_IMPLEMENTED);
             return;
         }
-        // TODO
+
+        List<SolrInputDocument> instructorDocs = new ArrayList<>();
+
+        for (InstructorAttributes instructor : instructors) {
+            SolrInputDocument instructorDoc = new InstructorSearchDocument(instructor).toDocument();
+            instructorDocs.add(instructorDoc);
+        }
+
+        addDocumentsToCollection(instructorDocs, INSTRUCTOR_COLLECTION_NAME);
     }
 
     /**
      * Removes instructor search documents based on the given keys.
      */
     public void deleteInstructorSearchDocuments(String... keys) {
-        if (!isSearchServiceActive()) {
-            log.warning("Search service is not implemented");
+        if (client == null) {
+            log.warning(ERROR_SEARCH_NOT_IMPLEMENTED);
+            return;
+        }
+
+        List<String> batchQueryString = Arrays.asList(keys);
+        if (batchQueryString.isEmpty()) {
+            return;
+        }
+
+        try {
+            client.deleteById(INSTRUCTOR_COLLECTION_NAME, batchQueryString);
+            client.commit(INSTRUCTOR_COLLECTION_NAME);
+        } catch (SolrServerException e) {
+            log.severe(String.format(ERROR_DELETE_DOCUMENT, batchQueryString, e.getRootCause())
+                    + TeammatesException.toStringWithStackTrace(e));
+        } catch (IOException e) {
+            log.severe(String.format(ERROR_DELETE_DOCUMENT, batchQueryString, e.getCause())
+                    + TeammatesException.toStringWithStackTrace(e));
+        }
+    }
+
+    /**
+     * Reset the data for all collections in search server.
+     * WARNING: should only be used for testing purposes.
```

Comment:
This does not align with the early return condition. `Config.isDevServer` does not necessarily translate to "testing". Also, even during local development (not when running the component tests), I don't think we'd want to allow resetting the search documents.

Reply:
You have a point. I still have one issue which is during local development, if we rely on a `boolean` flag in `build.properties` to tell if component tests are being run that would mean that the dev needs to constantly switch the `boolean` flag between `true` and `false`, whenever he wants to run component tests?

Reply:
That's because relying on `build.properties` is not the best way either

Reply:
File: `test.template.properties`
```
# This is the host URL for the full-text search service used by the system.
test.search.service.host=
```
Can we should put a warning here to inform devs working on search features to use a separate port from the one in `build.template.properties`? This is because if both development and test use the same port, and therefore the same Solr server, it is inevitable that running component tests will wipe out existing documents/collections used for development since they read and write to the same Solr server. 

Another reason for this proposal:  
By having two separate Solr server (one for development and one for test), we would no longer need to worry that running component tests will reset the data during local development.`!Config.isDevServer` would then suffice to ensure that this method can never be called in production. 

Appreciate your inputs @wkurniawan07 @Derek-Hardy

Reply:
The port is not the determining factor. I can, for instance, have two separate Solr instances (from different directories, from different Docker containers, etc.), and as long as the correct instance is used, the port doesn't matter. In fact, even if different port is used, if the Solr instance is the same then it will still wipe out the collection.

Reply:
`SearchManager` is only used in `InstructorsDb`, `StudentsDb` and tests.

How about:

**Function signature:** `public void resetCollections(Object caller)`

**Use:** `SearchManagerFactory.getSearchManager().resetCollections(this);`

**Check:** `if (!(caller instanceof EntitiesDb))`

Reply:
This might be a way to restrict the method only running in JUnit environment?
https://stackoverflow.com/a/12717377/9421552

Can't think of any project variables that does the work...

Reply:
Why not just a simple boolean flag within `SearchManager` itself?

Reply:
You mean having a field like `IS_SERVICE_UNDER_TEST = false` in the `SearchManager` class. You set it to true only before performing component tests? 

Then we might risk devs accidentally commit it. And it is up to their discretion then?

Reply:
Why make it a constant? It could be a field that is set during initialization.

Reply:
Added a `isResetAllowed` field initialised in constructor. During local development, the flag is `false` by default, which bans reset of collections. During local test, this flag will be initialised to `true`. In production environment, the reset code is completely banned from being run by checking `Config.isDevServer() && isResetAllowed`.